pkg/arvo/lib/hood/drum.hoon

```diff
@@ -225,51 +225,27 @@
   ==
 ::
 ++  on-load
-  |=  ver=?(%1 %2 %3)
-  ?-    ver
-      %1
-    =<  se-abet  =<  se-view
-    =<  (se-emit %pass /kiln %arvo %g %sear ~wisrut-nocsub)
-    =<  (se-born %home %goad)
-    =<  (se-born %home %metadata-store)
-    =<  (se-born %home %metadata-hook)
-    =<  (se-born %home %contact-store)
-    =<  (se-born %home %contact-hook)
-    =<  (se-born %home %contact-view)
-    =<  (se-born %home %link-store)
-    =<  (se-born %home %link-proxy-hook)
-    =<  (se-born %home %link-listen-hook)
-    =<  (se-born %home %link-view)
-    (se-born %home %s3-store)
-  ::
-      %2
-    =<  se-abet  =<  se-view
-    =<  (se-emit %pass /kiln %arvo %g %sear ~wisrut-nocsub)
-    =<  (se-born %home %metadata-store)
-    =<  (se-born %home %metadata-hook)
-    =<  (se-born %home %contact-store)
-    =<  (se-born %home %contact-hook)
-    =<  (se-born %home %contact-view)
-    =<  (se-born %home %link-store)
-    =<  (se-born %home %link-proxy-hook)
-    =<  (se-born %home %link-listen-hook)
-    =<  (se-born %home %link-view)
-    (se-born %home %s3-store)
-  ::
-      %3
-    =<  se-abet  =<  se-view
-    =<  (se-emit %pass /kiln %arvo %g %sear ~wisrut-nocsub)
-    =<  (se-born %home %metadata-store)
-    =<  (se-born %home %metadata-hook)
-    =<  (se-born %home %contact-store)
-    =<  (se-born %home %contact-hook)
-    =<  (se-born %home %contact-view)
-    =<  (se-born %home %link-store)
-    =<  (se-born %home %link-proxy-hook)
-    =<  (se-born %home %link-listen-hook)
-    =<  (se-born %home %link-view)
-    (se-born %home %s3-store)
-  ==
+  |=  ver=?(%1 %2 %3 %4)
+  =<  se-abet  =<  se-view
+  =?  .  (lte ver %3)
+    =.  ver  %4
+    =.  ..on-load
+      =<  (se-emit %pass /kiln %arvo %g %sear ~wisrut-nocsub)
+      =<  (se-born %home %goad)
+      =<  (se-born %home %metadata-store)
+      =<  (se-born %home %metadata-hook)
+      =<  (se-born %home %contact-store)
+      =<  (se-born %home %contact-hook)
+      =<  (se-born %home %contact-view)
+      =<  (se-born %home %link-store)
+      =<  (se-born %home %link-proxy-hook)
+      =<  (se-born %home %link-listen-hook)
+      =<  (se-born %home %link-view)
+      (se-born %home %s3-store)
+    .
+  ?>  ?=(%4 ver)
+  =>  (se-drop:(se-pull our.hid %dojo) | our.hid %dojo)
+  (se-drop:(se-pull our.hid %chat-cli) | our.hid %chat-cli)
```

Comment:
It's not clear to me how this _reconnects_, after pulling the existing subscription. Dojo automatically re-links in `+se-drop`, but don't you want a manual `se-link` here for chat-cli?

Reply:
`eel` is the set of apps we want to be connected to, and `fug` is the set of apps we believe we are connected to.  `+se-link` just puts it in `eel`, and later it gets linked in `+se-adze` (called in `+se-abet`).  `+se-drop` removes it from `fug`, and if the `pej` ("permanent") argument is yes, then it'll remove it from `eel` as well.  Since it's no, we don't, and it gets restarted.

`+se-pull` unsubscribes (which you don't need to do if you're unlinking because of a negative watch-ack, for example).  `+se-nuke` is almost exactly what I have except `pej` is yes.  `+se-drop` special cases dojo so that even if you run `|unlink %dojo`, you still resubscribee.

Reply:
Thanks for the explanation, this seems to check out!

Reply:
Should the 10 se-born apps be collected into a `:~`? For that matter, is there an explicit "apps that should be running" data structure this could be drawing from?